### PR TITLE
Enhance softlayer driver

### PR DESCRIPTION
**Pull request by wzymaster**
To enhance the driver, I added two new parameters to address two issues here :

- SOFTLAYER_DOCKER_ON_PRIVATE_IP, enforce docker daemon to run on the private ip if needed. Especially for some use cases in private cloud.

- SOFTLAYER_NETWORK_MAX_SPEED, set up network speed and the default is set to 100 Mbps. The SoftLayer driver in the release version only provision VM with 10 Mbps by the default, which is not good in the most cases for users.

### CODE REVIEW

**Comment by wzymaster** [comment]
I get the failure like below but don't understand why this error pops up : 

```
drivers/softlayer/driver.go
drivers/softlayer/softlayer.go
make: *** [fmt] Error 1
mk/validate.mk:12: recipe for target 'fmt' failed
make: *** [build validate] Error 2
```

Anyone can help ?

**Comment by nathanleclaire** [comment]
It wants `gofmt` run on the code.  Please do that and re-push

File: version/version.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 
 var (
 	// Version should be updated by hand at each release
-	Version = "0.7.0"
+	Version = "0.7.0.1"
```
Comment (nathanleclaire) [inline version/version.go]: This should not be changed (in this PR, looks like we should update it otherwise though ;) )

**Comment by wzymaster** [comment]
@nathanleclaire  Can you tell me what you mean 'gofmt' run on the code ? how can I do it ?

**Comment by nathanleclaire** [comment]
something like `gofmt -s -w **/*.go` -- you can probably mount machine source into a container to do it if you don't have go locally, something like `docker run -ti -w /machine -v $(pwd):/machine sh -c 'gofmt -s -w **/*.go'`

**Comment by wzymaster** [comment]
@nathanleclaire I am using container to build machine but I get nothing to push after successful running the cmd line provided by you like : 

```
Leos-MacBook-Pro-2:machine leowu$ docker run -ti -w /machine -v $(pwd):/machine golang:alpine sh -c 'gofmt -s -w **/*.go'
Unable to find image 'golang:alpine' locally
alpine: Pulling from library/golang
e110a4a17941: Already exists
ac58758e6ad5: Pull complete
de90fc0e5164: Pull complete
e8f38fc389bc: Pull complete
087a26f98166: Pull complete
25994cc11974: Pull complete
Digest: sha256:8b9e1e4a137e7663c1dc52a33c41699661ea4da9aca6ab0771b1fcec16a87535
Status: Downloaded newer image for golang:alpine
Leos-MacBook-Pro-2:machine leowu$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
```

**Comment by nathanleclaire** [comment]
shell glob might not be matching.  try 

```console
$ docker run -ti -w /machine -v $(pwd):/machine golang:alpine sh -c 'gofmt -s -w drivers/softlayer/*.go'
```

etc.

**Comment by wzymaster** [comment]
@nathanleclaire  sorry, I just fix the problem with `gofmt` but get a new error : 

```
mk/validate.mk:15: recipe for target 'vet' failed
make: *** [vet] Error 1
make: *** [build validate] Error 2
```

Please advise it.

**Comment by nathanleclaire** [comment]
Look in the Travis CI output.

```
drivers/softlayer/driver.go:375: unreachable code
```

Since [this section of the code](https://github.com/docker/machine/pull/3654/files#diff-50b71984e556cb76a3e64ec0009d43f5R368) always exits, you can't reach the rest.  Something is off in that function, for sure.

**Comment by wzymaster** [comment]
@nathanleclaire  what's next after all checks have passed ?

**Comment by nathanleclaire** [comment]
LGTM

**Comment by nathanleclaire** [comment]
Thanks for the contribution @wzymaster

**Comment by wzymaster** [comment]
@nathanleclaire  just a quick question, this change will go into 0.9.0 release of Docker Machine ?

**Comment by nathanleclaire** [comment]
Yes

**Pull request merged**